File: Data Structures & Algorithms/max-water-container/submission-1.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        N = len(heights)

        i = 0
        j = N-1

        ans = 0
        while i < j:
            ans = max(ans, min(heights[i], heights[j]) * (j-i) )

            if heights[i] == heights[j]:
                i += 1
                j -= 1
            elif heights[i] > heights[j]:
                j -= 1
            else:
                i += 1
        return ans

File: Data Structures & Algorithms/insert-new-interval/submission-1.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> updatedIntervals;
        for(const auto& interval : intervals) {
            if (interval[1] < newInterval[0]) {
                updatedIntervals.push_back(interval);
            }
            else if (newInterval[1] < interval[0]) {
                updatedIntervals.push_back(newInterval);
                newInterval = interval;
            }
            else {
                newInterval = {min(newInterval[0], interval[0]), max(newInterval[1], interval[1])};
            }
        }
        updatedIntervals.push_back(newInterval);
        return updatedIntervals;
    }
};
